### ISSUE TITLE
self.method_missing on require respond_to_missing as instance method instead of class method

### ISSUE THREAD

**Issue by Rataah** [bbatsov/rubocop#3358]
Hi,

For some reasons, I defined the `method_missing` as a **class method**. 
The corresponding cop rejects my code because `respond_to_missing?` is missing. 
I assume this method should be a **class method** too but this doesn't work. The cop accepts only if I declare an **instance method**.

Here is an example:
```ruby
# Rubocop method_missing issue
# this should be OK
class GoodExample
  def self.some_method
  end

  def self.method_missing(method, *args)
    super
  end

  def self.respond_to_missing?
    true
  end
end

# Rubocop method_missing issue
# this works but not as we expect
class BadExample
  def self.some_method
  end

  def self.method_missing(method, *args)
    super
  end

  def respond_to_missing?
    true
  end
end
```
```shell
Inspecting 1 file
C

Offenses:

example.rb:7:3: C: When using method_missing, define respond_to_missing?.
  def self.method_missing(method, *args) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Version used: 0.42.0

**Comment by kke** [comment]
Pushing to rubygems anytime soon?
